### ISSUE TITLE
casper client rs deployhash is conflicting with node's deployhash

### ISSUE THREAD

**Issue by RitaMAllenCA** [casper-ecosystem/casper-client-rs#50]
running into an issue with the casper-client-rs. During development for checkpoint charlie I ran into type conflicts between the nodes DeployHash and clients DeployHash.

**Comment by andrzej-casper** [comment]
As a workaround you can use older `casper-node` version:
![image](https://user-images.githubusercontent.com/121791569/222915046-68b35e5c-689d-43e0-a152-5325648656ea.png)

**Comment by andrzej-casper** [comment]
Full compilation error below:

```
error[E0308]: arguments to this function are incorrect
   --> lib/deploy.rs:235:22
    |
235 |         let deploy = Deploy::new(
    |                      ^^^^^^^^^^^
...
241 |             payment,
    |             ------- expected enum `casper_execution_engine::core::engine_state::executable_deploy_item::ExecutableDeployItem`, found enum `ExecutableDeployItem`
242 |             session,
    |             ------- expected enum `casper_execution_engine::core::engine_state::executable_deploy_item::ExecutableDeployItem`, found enum `ExecutableDeployItem`
    |
    = note: enum `ExecutableDeployItem` and enum `casper_execution_engine::core::engine_state::executable_deploy_item::ExecutableDeployItem` have similar names, but are actually distinct types
note: enum `ExecutableDeployItem` is defined in crate `casper_execution_engine`
   --> /home/andrew/.cargo/registry/src/github.com-1ecc6299db9ec823/casper-execution-engine-2.0.1/src/core/engine_state/executable_deploy_item.rs:111:1
    |
111 | pub enum ExecutableDeployItem {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: enum `casper_execution_engine::core::engine_state::executable_deploy_item::ExecutableDeployItem` is defined in crate `casper_execution_engine`
   --> /home/andrew/.cargo/registry/src/github.com-1ecc6299db9ec823/casper-execution-engine-3.0.0/src/core/engine_state/executable_deploy_item.rs:111:1
    |
111 | pub enum ExecutableDeployItem {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `casper_execution_engine` are being used?
    = note: enum `ExecutableDeployItem` and enum `casper_execution_engine::core::engine_state::executable_deploy_item::ExecutableDeployItem` have similar names, but are actually distinct types
note: enum `ExecutableDeployItem` is defined in crate `casper_execution_engine`
   --> /home/andrew/.cargo/registry/src/github.com-1ecc6299db9ec823/casper-execution-engine-2.0.1/src/core/engine_state/executable_deploy_item.rs:111:1
    |
111 | pub enum ExecutableDeployItem {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: enum `casper_execution_engine::core::engine_state::executable_deploy_item::ExecutableDeployItem` is defined in crate `casper_execution_engine`
   --> /home/andrew/.cargo/registry/src/github.com-1ecc6299db9ec823/casper-execution-engine-3.0.0/src/core/engine_state/executable_deploy_item.rs:111:1
    |
111 | pub enum ExecutableDeployItem {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `casper_execution_engine` are being used?
note: associated function defined here
   --> /home/andrew/.cargo/registry/src/github.com-1ecc6299db9ec823/casper-node-1.4.13/src/types/deploy.rs:647:12
    |
647 |     pub fn new(
    |            ^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `casper-client` due to previous error
```

There is suggestion about two different versions of crate `casper_execution_engine` being used, so let us check `Cargo.lock`:

```
[[package]]
name = "casper-client"
version = "1.5.0"
dependencies = [
 ...
 "casper-execution-engine 2.0.1",
]

...

[[package]]
name = "casper-node"
version = "1.4.13"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "055ec2b61e829acf813fdda0b5ebd82c40b3f46800fc9098f6fa78a103845108"
dependencies = [
 ...
 "casper-execution-engine 3.0.0",
]
```

That is right! Rust :heart:.

**Comment by andrzej-casper** [comment]
It was fixed in https://github.com/casper-ecosystem/casper-client-rs/commit/e421e81506c64b020d2f06fe22e45a7342b246f8, which is part of the newly released `v1.5.1`.
